### PR TITLE
Quality: improve typing via plugwise v0.31.3

### PR DESCRIPTION
**Pull request by bouwew**

### CODE REVIEW

**Review by CoMPaTech — DISMISSED** [review]

**Comment by CoMPaTech** [comment]
@bouwew please check TODO section in Models wrt existence (and forthcoming improvement of output) of [3a95d72](https://github.com/plugwise/plugwise-beta/pull/397/commits/3a95d7216cac6733cc4b6edd5876dc87a7d6766e)

**Comment by CoMPaTech** [comment]
Adds 4 TODOs wrt the fixtures vs models vs tests

And as predicted by bouwew also re-introduces new mypy failures.

On the bright side, no more manual fixtures, we have a generator for that now in https://github.com/plugwise/python-plugwise/pull/308

**Comment by CoMPaTech** [comment]
Note on 518c54c still needs `max_dhw_temperature` change in https://github.com/plugwise/python-plugwise/pull/309/files to be applied (i.e. python-plugwise generates this fixture but test in number-domain fails). `prettier` should now no longer tinker with the generated fixtures from https://github.com/plugwise/python-plugwise/pull/311

File: custom_components/plugwise/number.py [hunk 1]
```diff
@@ -14,32 +15,31 @@
...
-    NumberEntityDescription, PlugwiseEntityDescriptionMixin
-):
+class PlugwiseNumberEntityDescription(NumberEntityDescription, PlugwiseNumberMixin):
     """Class describing Plugwise Number entities."""
 
-    native_max_value_key: str | None = None
-    native_min_value_key: str | None = None
-    native_step_key: str | None = None
-    native_value_key: str | None = None
-
```
Comment (bouwew) [inline custom_components/plugwise/number.py:42]: Please change back using native in the varnames as indicated in comment on Discord.

Comment (CoMPaTech) [reply custom_components/plugwise/number.py:42]: 👍

**Comment by CoMPaTech** [comment]
max_dhw not introduced in Core PR yet (as that would be new functionality + involve new translations)

**Review by CoMPaTech — APPROVED** [review]
Re-added dhw now (so undoing stale - though it would be approving my own latest commits this way) :)

**Review by CoMPaTech — APPROVED** [review]

**Pull request merged**